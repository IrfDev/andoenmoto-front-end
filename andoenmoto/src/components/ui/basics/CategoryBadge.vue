<template>
  <div
    class="flex-nowrap d-flex justify-content-end"
    :class="{
      'flex-column-reverse': reverse,
      'flex-column align-content-stretch': !reverse,
      'col-12': name === 'Motocicletas',
      'col-md-3 col-4': name !== 'Motocicletas',
    }"
    @click="activateCategoryy($attrs.category)"
  >
    <div class="badge-image m-0">
      <img :src="image" :alt="name" class="img-fluid w-50" />
    </div>
    <div class="badge-title m-0">
      <h2 v-if="reverse">{{ name }}</h2>
      <h3 v-else>{{ name }}</h3>
    </div>
  </div>
</template>

<script>
export default {
  name: 'CategoryBadge',
  methods: {
    async activateCategoryy(category) {
      if (this.$route.name !== 'categories-style') {
        await this.$store.dispatch('categories/activateCategory', category);
        this.$router.push({
          name: 'categories-style',
          params: { category: category.name },
        });
      }
    },
  },
  props: {
    name: {
      type: String,
      required: true,
    },
    image: {
      type: String,
      required: true,
    },
    colWidth: {
      type: Number,
    },
    reverse: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style lang="scss" scoped>
h3,
h2 {
  color: $alpha-white;
  font-family: $title;
}
.flex-column-reverse {
  h3,
  h2 {
    font-family: $typo !important;
    color: $alpha-white;
  }
}
</style>
