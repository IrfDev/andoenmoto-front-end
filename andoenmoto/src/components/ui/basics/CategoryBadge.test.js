import { mount } from '@vue/test-utils';
import CategoryBadge from './CategoryBadge.vue';

const createCmp = (name, image) => mount(CategoryBadge, {
  propsData: {
    name,
    image,
  }
});

describe('Category Badge Snapshot', () => {
  it('Props are corrects', () => {
    const cmp = createCmp('Motos', '/motos.png');
    expect(cmp.element).toMatchSnapshot();
  });
});