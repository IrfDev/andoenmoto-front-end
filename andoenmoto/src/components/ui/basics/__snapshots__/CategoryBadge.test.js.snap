// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Badge Snapshot Props are corrects 1`] = `
<div
  class="col-undefined flex-nowrap d-flex flex-column align-content-stretch"
>
  <div
    class="badge-image m-0"
  >
    <img
      alt="Motos"
      class="img-fluid"
      src="/motos.png"
    />
  </div>
   
  <div
    class="badge-title m-0"
  >
    <h4>
      Motos
    </h4>
  </div>
</div>
`;
